nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d_ros.hpp

```diff
@@ -134,6 +134,42 @@ class Costmap2DROS : public nav2_util::LifecycleNode
    */
   nav2_util::CallbackReturn on_shutdown(const rclcpp_lifecycle::State & state) override;
 
+  /**
+   * @brief child-LifecycleNode mode
+   * sometimes costmap may be launched by another LifecycleNode and work as a child-thread
+   * child-LifecycleNode should react to ctrl+C later than its parents to avoid unclean shutdown
+   * Thus, it's neccessary to turnChildLifecycleNode() to change its launch mode
+   *
+   * in ChildLifecycleNode mode, it would react to rcl_preshutdown anymore
+   * all its lifecycle state would be controlled by its parent-LifecycleNode
+   */
+  void turnChildLifecycleNode()
```

Comment:
Better option:
- https://github.com/ros-planning/navigation2/blob/4bc3e8a534005e983ee7a49a09ec00aeb9820661/nav2_costmap_2d/src/costmap_2d_node.cpp#L46

This non-argument constructor is only used in the node, not in any of the internal uses of costmap 2d. [Update it to take in NodeOptions with a default constructed option](https://github.com/ros-planning/navigation2/blob/4bc3e8a534005e983ee7a49a09ec00aeb9820661/nav2_planner/include/nav2_planner/planner_server.hpp#L58).

Then, we know that constructor is only used in the `Node` or `Component` node independent operation. Any internal-node will use the other constructors. That eliminates `turnChildLifecycleNode`. 

In this constructor set `is_lifecycle_follower_ = false;`

Reply:
I'm not very clear about how to do it, while I've tried to modify our code in our latest change.